common/build/eslint-plugin-fluid/src/test/mockFiles/no-unchecked-record-access/fileWithOnlyRecordAccess.ts

```diff
@@ -0,0 +1,84 @@
+someObjWithPotenciallyUndefinedProperties.a.length; // defect: Accessing length of dynamic property 'a', TypeScript can't guarantee 'a' is a string
+someObjWithPotenciallyUndefinedProperties["a"]; // ok: Accessing property 'a' using bracket notation
+someObjWithPotenciallyUndefinedProperties["a"].length; // defect: Accessing length of potencially undefined property 'a' using bracket notation
+someObjWithPotenciallyUndefinedProperties[a].length; // ok: Accessing length of string property 'a' using bracket notation
+someObjWithPotenciallyUndefinedProperties[b].length; // ok: Accessing length of string property 'b' using bracket notation
+someObjWithPotenciallyUndefinedProperties[c].length; // defect: Accessing length of undefined property 'c' using bracket notation
+someObjWithPotenciallyUndefinedProperties.a?.length; // ok: Using optional chaining to access length, safely handles 'undefined'
+
+if (someObjWithPotenciallyUndefinedProperties.a) {
+	someObjWithPotenciallyUndefinedProperties.a.length; // ok: Within a truthy check, 'a' is guaranteed to be defined
+}
+
+for (const key in someObjWithPotenciallyUndefinedProperties) {
+	someObjWithPotenciallyUndefinedProperties[key]; // ok: Accessing dynamic property using 'in'
+	someObjWithPotenciallyUndefinedProperties[key].length; // ok: Accessing dynamic property using 'in'
+}
+
+for (const [key, value] of Object.entries(someObjWithPotenciallyUndefinedProperties)) {
+	value; // ok: Accessing value from entries
+	value.length; // ok: Accessing length of value from entries, entries should be string values
+	someObjWithPotenciallyUndefinedProperties[key]; // ok: Accessing dynamic property using 'in'
+	someObjWithPotenciallyUndefinedProperties[key].length; // ok: Accessing dynamic property using 'in'
+}
+
+const aExpectingString: string = someObjWithPotenciallyUndefinedProperties.a; // defect: Assigning dynamic property to a strict string variable, 'a' might be undefined
+const aExpectingStringOrUndefined: string | undefined = someObjWithPotenciallyUndefinedProperties.a; // ok: Assigning dynamic property to a string or undefined variable, 'a' might be undefined
+const aImplicitType = someObjWithPotenciallyUndefinedProperties.a; // ok: Assigning dynamic property with inferred type
+aImplicitType.length; // defect: Accessing length of inferred type, 'a' might be undefined
+aImplicitType?.length; // ok: Optional chaining to access length, safely handles 'undefined'
+
+if ("a" in someObjWithPotenciallyUndefinedProperties) {
+	someObjWithPotenciallyUndefinedProperties.a.length; // ok: Accessing length of property inside an 'in' check, 'a' is guaranteed to be defined
+}
+
+interface NonNullableProps {
+	definitelyString: string;
+	maybeString?: string;
+}
+
+const nonNullObj: NonNullableProps = { definitelyString: "hello" };
+nonNullObj.definitelyString.length; // ok: Accessing length of non-nullable property
+nonNullObj.maybeString.length; // defect: Accessing length of potentially undefined property
```

Comment:
While this is a defect, it is not something I expect the linter case to check for. There isn't an index access here. So, by linter checks, this should be "ok" (no defect). Tsc complains if this is compiled, right?

Reply:
Yes tsc does complain even when noUIA is disabled